implementations/rs/ockam/src/contact/contact.rs

```diff
@@ -0,0 +1,39 @@
+use crate::contact::contact_event::ContactEvent;
+use crate::contact::ContactVault;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+pub type ContactTags = HashMap<String, String>;
+
+#[derive(Clone)]
+pub struct Contact {
+    identifier: String,
+    events: Vec<ContactEvent>,
+    vault: Arc<Mutex<dyn ContactVault>>,
+}
+
+impl Contact {
+    pub(crate) fn new(
+        identifier: String,
+        events: Vec<ContactEvent>,
+        vault: Arc<Mutex<dyn ContactVault>>,
+    ) -> Self {
+        Contact {
+            identifier,
+            events,
+            vault,
+        }
+    }
+}
+
+impl Contact {
+    pub fn identifier(&self) -> &str {
+        &self.identifier
+    }
+    pub fn events(&self) -> &Vec<ContactEvent> {
```

Comment:
```suggestion
    pub fn events(&self) -> &[ContactEvent] {
```

Reply:
@mikelodder7 I wonder why returning slice is better

Reply:
The only case coming up to my mind - it simplifies things when we want to use Vec vs something heapless depending on enabled features